registry/keda-scaler.yaml

```diff
@@ -0,0 +1,41 @@
+# Code generated by KubeVela templates. DO NOT EDIT.
+apiVersion: core.oam.dev/v1beta1
+kind: TraitDefinition
+metadata:
+  name: keda-scaler
+  annotations:
+    definition.oam.dev/description: "keda-scaler supports multiple event to elastically scale applications, this scaler only applies to ScaledObject"
+spec:
+  definitionRef:
+    name: scaledobjects.keda.sh
+  appliesToWorkloads:
+    - deployments.apps
+  schematic:
+    cue:
+      template: |
+        outputs: kedaScaler: {
+        	apiVersion: "keda.sh/v1alpha1"
+        	kind:       "ScaledObject"
+        	metadata: {
+        		name: context.name
+        	}
+        	spec: {
+        		scaleTargetRef: {
+        			name: context.name
+        		}
+        		triggers: [{
+        			type: parameter.type
+        			metadata: {
+        				type:  "Utilization"
+        				value: parameter.value
```

Comment:
Here are two types, let's make it more clear. You can refer to https://github.com/oam-dev/catalog/blob/70d63115f238b01dd76f3898d6ca6efce4fe66c5/traits/autoscalertrait/api/v1alpha1/autoscaler_types.go#L64-L73. @wonderflow and I discussed it several times.

The reason why you are encouraged to modify it is the spec of keda and HPA are mixed-up, sometime confusing.

Reply:
got it